common-utils/src/main/java/eu/arrowhead/common/mqtt/ArrowheadMqttService.java

```diff
@@ -0,0 +1,112 @@
+import org.apache.logging.log4j.Logger;
+import org.eclipse.paho.client.mqttv3.MqttClient;
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.eclipse.paho.client.mqttv3.MqttMessage;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Service;
+import org.springframework.util.Assert;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import eu.arrowhead.common.Constants;
+import eu.arrowhead.common.Utilities;
+import eu.arrowhead.common.exception.ExternalServerError;
+import eu.arrowhead.common.exception.InternalServerError;
+import eu.arrowhead.dto.MqttResponseTemplate;
+
+@Service
+@ConditionalOnProperty(name = Constants.MQTT_API_ENABLED, matchIfMissing = false)
+public class ArrowheadMqttService {
+
+	//=================================================================================================
+	// members
+
+	@Autowired
+	private MqttService mqttService;
+
+	@Autowired
+	private ObjectMapper mapper;
+
+	private final Logger logger = LogManager.getLogger(getClass());
+
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------
+	/**
+	 * Subscribe for consuming a push service
+	 */
+	public void subscribe(final String connectId, final String topic, final MqttQoS qos) {
+		logger.debug("subscribe started");
+		Assert.isTrue(!Utilities.isEmpty(connectId), "connectId is empty");
+		Assert.isTrue(!Utilities.isEmpty(topic), "topic is empty");
+
+		// TODO
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	/**
+	 * Unsubscribe from consuming a push service
+	 */
+	public void unsubscribe(final String connectId, final String[] topics) {
+		logger.debug("unsubscribe started");
+		Assert.isTrue(!Utilities.isEmpty(connectId), "connectId is empty");
+		Assert.notNull(topics, "topic array is null");
+
+		final MqttClient client = mqttService.client(connectId);
+		Assert.notNull(client, "Unknown connectId: " + connectId);
+
+		try {
+			client.unsubscribe(topics);
+		} catch (final MqttException ex) {
+			logger.debug(ex);
+			throw new ExternalServerError("MQTT unsubscribe failed: " + ex.getMessage());
+		}
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	/**
+	 * Publish a service message
+	 */
+	public void publish(final String connectId, final String sender, final String topic, final MqttQoS qos, final Object payload) {
+		logger.debug("publish started");
+		Assert.isTrue(!Utilities.isEmpty(connectId), "connectId is empty");
+		Assert.isTrue(!Utilities.isEmpty(topic), "topic is empty");
+
+		// TODO needs an MqttPublishTemplate
```

Comment:
it would be wise to create an issue about this TODO, just to not forget to implement it.